src/borg/remote.py

```diff
@@ -755,8 +766,11 @@ def handle_error(unpacked):
                             raise UnexpectedRPCDataFormatFromServer(data)
                         if msgid in self.ignore_responses:
                             self.ignore_responses.remove(msgid)
+                            # async methods never return values, but may raise exceptions.
                             if b'exception_class' in unpacked:
-                                handle_error(unpacked)
```

Comment:
the `delete --force` case is explicitly not about rolling back on error.
`delete --force` uses a `on error resume next` policy. Because at that point repository inconsistencies are expected and it should just get rid of anything that is left.
I think the old behavior is ok when combined with rollback (just maybe not giving good error messages).